version/version.go

```diff
@@ -17,14 +17,117 @@
 // Package version represents the version the project Dragonfly.
 package version
 
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"runtime"
+	"strings"
+	"text/template"
+)
+
 var (
-	// Version is the version of the project Dragonfly
+	// version is the version of project Dragonfly
+	// populate via ldflags
+	version string
+
+	// revision is the current git commit revision
+	// populate via ldflags
+	revision string
+
+	// branch is the current git branch
+	// populate via ldflags
+	branch string
+
+	// buildDate is the build date of project Dragonfly
 	// populate via ldflags
-	Version string
+	buildDate string
+
+	// goVersion is the version of Golang
+	goVersion = runtime.Version()
 
 	// DFDaemonVersion is the version of dfdaemon
-	DFDaemonVersion = Version
+	DFDaemonVersion = version
 
 	// DFGetVersion is the version of dfget
-	DFGetVersion = Version
+	DFGetVersion = version
+
+	// SupernodeVersion is the version of supernode
+	SupernodeVersion = version
+
+	// DFVersion the global instance of DragonflyVersion
+	DFVersion *DragonflyVersion
 )
+
+func init() {
+	DFVersion = &DragonflyVersion{
+		version,
+		revision,
+		branch,
+		buildDate,
+		goVersion,
+	}
+}
+
+// versionInfoTmpl contains the template used by Info.
+var versionInfoTmpl = `
+{{.program}}, version  {{.version}}
+  git  branch:      {{.branch}}
+  git  commit:      {{.revision}}
+  build  date:      {{.buildDate}}
+  go  version:      {{.goVersion}}
+`
+
+// Print returns version information.
+func Print(program string) string {
+	m := map[string]string{
+		"program":   program,
+		"version":   version,
+		"branch":    branch,
+		"revision":  revision,
+		"buildDate": buildDate,
+		"goVersion": goVersion,
+	}
+	t := template.Must(template.New("version").Parse(versionInfoTmpl))
+
+	var buf bytes.Buffer
+	if err := t.ExecuteTemplate(&buf, "version", m); err != nil {
+		panic(err)
+	}
+	return strings.TrimSpace(buf.String())
+}
+
+// DragonflyVersion contains build information about Dragonfly.
+type DragonflyVersion struct {
+	Version   string `json:"version"`
+	Revision  string `json:"revision"`
+	Branch    string `json:"branch"`
+	BuildDate string `json:"buildDate"`
+	GoVersion string `json:"goVersion"`
+}
+
+// Handler returns build information
+func Handler(w http.ResponseWriter, r *http.Request) {
```

Comment:
Extract the common code of  the functions  `Handler` and `HandlerWithCtx`.

Reply:
Done.